app/services/news-banner.ts

```diff
@@ -0,0 +1,85 @@
+import { StatefulService, mutation } from './stateful-service';
+import { UserService } from './user';
+import { HostsService } from './hosts';
+import { Inject } from '../util/injector';
+import { authorizedHeaders } from '../util/requests';
+
+interface INewsBannerInfo {
+  id: number;
+  header: string;
+  subHeader: string;
+  link: string;
+  linkTitle: string;
+  thumbnail: string;
+}
+
+export class NewsBannerService extends StatefulService<INewsBannerInfo> {
+  @Inject() private hostsService: HostsService;
+  @Inject() private userService: UserService;
+
+  static initialState: INewsBannerInfo = {
+    id: null,
+    header: '',
+    subHeader: null,
+    link: null,
+    linkTitle: null,
+    thumbnail: null
+  };
+
+  async updateBanner() {
+    const newBanner = await this.fetchBanner();
+    this.SET_BANNER(newBanner);
+  }
+
+  bannerExists() {
+    return this.state.id !== null;
+  }
+
+  closeBanner() {
+    this.postBannerClose();
+  }
+
+  private async fetchBanner() {
+    const endpoint = `api/v5/slobs/announcement/get?clientId=${this.userService.getLocalUserId()}`
+    const req = this.formRequest(endpoint);
+    try {
+      const newState = await fetch(req).then((rawResp) => rawResp.json());
+      return newState.id ? newState : this.state;
+    } catch (e) {
+      return this.state;
+    }
+  }
+
+  private async postBannerClose() {
+    const endpoint = 'api/v5/slobs/announcement/close';
+    const postData = {
+      method: 'POST',
+      body: JSON.stringify({ clientId: this.userService.getLocalUserId(), announcementId: this.state.id }),
+      headers: new Headers({ 'Content-Type': 'application/json' })
+    };
+    const req = this.formRequest(endpoint, postData);
+    try {
+      await fetch(req);
+      this.CLEAR_BANNER();
+    } catch (e) {
+
+    }
+  }
+
+  private formRequest(endpoint: string, options: any = {}) {
+    const host = this.hostsService.streamlabs;
+    const headers = authorizedHeaders(this.userService.apiToken, options.headers);
```

Comment:
Do we show news only for authorized users?

Reply:
I don't think the spec said so, but I'll clarify with Eric

Reply:
@holiber it should be able to work for non-logged in users too, is there a way to generate auth-ed headers without a logged in user?

Reply:
There is no way to do that because it can cause serious security issues.
If we want to show this banner for unauthorized users we must rid of authorization for `https://streamlabs.com/api/v5/slobs/announcement/get`

However as I see you are disabling this banner by calling an API request. That makes me feel this API is not intended to work with unauthorized clients. I think you need to ask somebody from stemlabs backend team to resolve this or show this banner only for authorized users

Reply:
confirmed that its fine for only logged in users